inst/tests/tests.Rraw

```diff
@@ -21192,3 +21192,16 @@ test(2319.1, !is.null(attr(dt_get, ".internal.selfref")))
 dt_get0 = data.frame(a = 1:3, b = letters[1:3])
 setDT(get0("dt_get0"))
 test(2319.2, !is.null(attr(dt_get0, ".internal.selfref")))
+
+# Improved fread error handling for cmd exe and decompression #5415
+test(2320.1, fread(cmd="false"), error="External command failed with exit code") # external command execution error handling
+
+if (test_R.utils) {
+  tmp <- tempfile(fileext=".csv.gz")
+  file.create(tmp)
+  conn <- file(tmp, 'wb')
+  writeBin(as.raw(c(31, 139)), conn)
```

Comment:
one last thing: please comment about these seemingly arbitrary numbers for future readers